pallets/nodle-staking/src/lib.rs

```diff
@@ -193,6 +195,52 @@ pub mod pallet {
             Self::deposit_event(Event::TotalSelectedSet(old, new));
             Ok(().into())
         }
+        #[pallet::weight(T::WeightInfo::set_staking_limits())]
+        pub fn set_staking_limits(
+            origin: OriginFor<T>,
+            max_stake_validators: u32,
+            min_stake_session_selection: BalanceOf<T>,
+            min_validator_bond: BalanceOf<T>,
+            min_nominator_total_bond: BalanceOf<T>,
+            min_nominator_chill_threshold: BalanceOf<T>,
+        ) -> DispatchResultWithPostInfo {
+            T::CancelOrigin::try_origin(origin)
+                .map(|_| ())
+                .or_else(ensure_root)?;
+
+            let old = <StakingMaxValidators<T>>::get();
+            <StakingMaxValidators<T>>::set(max_stake_validators);
+            Self::deposit_event(Event::StakingMaxValidators(old, max_stake_validators));
+
+            let old = <StakingMinStakeSessionSelection<T>>::get();
+            <StakingMinStakeSessionSelection<T>>::set(min_stake_session_selection);
+            Self::deposit_event(Event::StakingMinStakeSessionSelection(
+                old,
+                min_stake_session_selection,
+            ));
+
+            let old = <StakingMinValidatorBond<T>>::get();
+            <StakingMinValidatorBond<T>>::set(min_validator_bond);
+            Self::deposit_event(Event::StakingMinValidatorBond(old, min_validator_bond));
+
+            let old = <StakingMinNominatorTotalBond<T>>::get();
+            <StakingMinNominatorTotalBond<T>>::set(min_nominator_total_bond);
+            Self::deposit_event(Event::StakingMinNominatorTotalBond(
+                old,
+                min_nominator_total_bond,
+            ));
+
+            let old = <StakingMinNominationChillThreshold<T>>::get();
+            <StakingMinNominationChillThreshold<T>>::set(min_nominator_chill_threshold);
+            Self::deposit_event(Event::StakingMinNominationChillThreshold(
+                old,
+                min_nominator_chill_threshold,
+            ));
```

Comment:
We can probably have one big event for the all the function instead of triggering lots of them

Reply:
Done :+1: